### PR TITLE
Docs/0.6.0

### PR DESCRIPTION
**Pull request by drewbanin**
### Sample config files:
- dbt_project.yml
- profiles.yml

### Docs
- Archival
- Context variables (`this`, `target`, `run_started_at`, `invocation_id`) + `{{ vars(...) }}`
- `on_run_start` and `on_run_end`
- Remove verbiage around BigQuery
- Macros
- `--profiles-dir`
- Data testing
- `s/run-target/target/g`
- `--non-destructive` and `--full-refresh`

Also, fixed some broken links

### CODE REVIEW

**Comment by drewbanin** [comment]
TODO: Update http://dbt.readthedocs.io/en/docs-0.6.0/guide/upgrading/ w/ brew instructions

**Comment by jthandy** [comment]
Amaaazing. Made a bunch of really small tweaks, now LGTM 👍

File: docs/guide/upgrading.md [hunk 1]
```diff
@@ -1,13 +0,0 @@
-# Upgrading
```
Comment (drewbanin) [inline docs/guide/upgrading.md]: Eek! We need this file -- it shows up in the dbt output if your version is out-of-date:

```bash
$ dbt --verison
installed version: 0.6.0
   latest version: 0.6.1
Your version of dbt is out of date! You can find instructions for upgrading here:
http://dbt.readthedocs.io/en/master/guide/upgrading/
```

Comment (drewbanin) [inline docs/guide/upgrading.md]: I'll add this back in

**Comment by jthandy** [comment]
heh ok. the instructions are out of date though...you go through it better in setup at this point. i don't really care about this too much but it really doesn't make sense to have this its own thing and then to go over the same thing somewhere else.

**Comment by drewbanin** [comment]
@jthandy merging

**Pull request merged**